src/SFA.DAS.ApplyService.Application/Apply/Oversight/RecordOversightOutcomeHandler.cs

```diff
@@ -32,43 +32,102 @@ public async Task<bool> Handle(RecordOversightOutcomeCommand request, Cancellati
         {
             _logger.LogInformation($"Recording Oversight review status of {request.OversightStatus} for application Id {request.ApplicationId}");
 
-            var application = await _applyRepository.GetApplication(request.ApplicationId);
+            _auditService.StartTracking(UserAction.RecordOversightOutcome, request.UserId, request.UserName);
+
+            var application = await GetApplicationOrThrow(request.ApplicationId);
+
+            var (oversightReview, isNew) = await GetExistingOrNewOversightReview(request.ApplicationId);
+
+            ApplyChanges(oversightReview, request, isNew);
+            await SaveChanges(oversightReview, application, isNew);
+            
+            await _auditService.Save();
+
+            return true;
```

Comment:
Would a better check be asserting that at least 1 row was affected on SaveChanges?